### PR TITLE
Add yurt-manager controller framework

### PR DESCRIPTION
**Pull request by kadisi**
#### What type of PR is this?
/kind feature

#### Which issue(s) this PR fixes:

ref #1067

#### Does this PR introduce a user-facing change?

```release-note
add yurt-manager framework
```

### CODE REVIEW

File: hack/make-rules/add_controller.sh [hunk 1]
```diff
@@ -0,0 +1,750 @@
...
+      usage
+      ;;
+    *)
+      usage
+      ;;
+    esac
+done
+
+if [ -z $GROUP ] || [ -z $VERSION ] || [ -z $KIND ] || [ -z $SCOPE ] || [ -z $SHORTNAME ] ; then
+    usage
```
Comment (rambohe-ch) [inline hack/make-rules/add_controller.sh:82]: we need to exit after `usage`

Comment (kadisi) [reply hack/make-rules/add_controller.sh:82]: `exit 1` has exist in usage() function

File: hack/make-rules/add_controller.sh [hunk 2]
```diff
@@ -0,0 +1,764 @@
...
+    mkdir -p ${CRD_VERSION_DIR}
+
+    cat > $doc_file <<EOF
+$(create_header ${VERSION})
+EOF
+
+    cat > $group_version_info_file <<EOF
+$(create_header ${VERSION})
+
+// Package v1beta1 contains API Schema definitions for the apps v1beta1 API group
```
[the remark this answers is not in the document]
Comment (kadisi) [reply hack/make-rules/add_controller.sh]: good, done

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**